tests/integration/main_test.go

```diff
@@ -73,13 +77,35 @@ func TestOryJwt(t *testing.T) {
 	runTestsuite(t, ts, config)
 }
 
+func TestGateway(t *testing.T) {
+	config := testcontext.GetConfig()
+	ts, err := testcontext.New(config, gateway.NewTestsuite)
+	if err != nil {
+		t.Fatalf("Failed to create Gateway testsuite %s", err.Error())
+	}
+	defer ts.TearDown()
+	runTestsuite(t, ts, config)
+}
+
 func runTestsuite(t *testing.T, testsuite testcontext.Testsuite, config testcontext.Config) {
 	opts := createGoDogOpts(t, testsuite.FeaturePath(), config.TestConcurrency)
 	suite := godog.TestSuite{
 		Name: testsuite.Name(),
 		// We are not using ScenarioInitializer, as this function only needs to set up global resources
 		TestSuiteInitializer: func(ctx *godog.TestSuiteContext) {
+			ctx.BeforeSuite(func() {
+				if err := createApiGatewayCR(config); err != nil {
```

Comment:
shouldn't we do this part of the scenarios, like for istio?, e.g.:
https://github.com/kyma-project/istio/blob/66911ce5bb2d2dfde90fb1401f377868769ca9b9/tests/integration/features/istio/evaluation/istio_installation_demo.feature#L2-L6

tearing down with the hooks approach as well:
https://github.com/kyma-project/istio/blob/66911ce5bb2d2dfde90fb1401f377868769ca9b9/tests/integration/hooks.go#L18-L69